lms/djangoapps/courseware/access.py

```diff
@@ -136,8 +136,9 @@ def has_access(user, action, obj, course_key=None):
     if not user:
         user = AnonymousUser()
 
-    if in_preview_mode():
-        if not bool(has_staff_access_to_preview_mode(user=user, obj=obj, course_key=course_key)):
+    # Preview mode is only accessible by staff.
+    if in_preview_mode() and course_key:
```

Comment:
When a course run is being accessed, this method is called with `course_key` to check for access to this course. ([Source](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/access.py#L128-L128)). 
It will not be possible to be in preview mode without `course_key`.